flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java

```diff
@@ -542,6 +551,60 @@ private Operation convertAlterTableReset(
         return new AlterTableOptionsOperation(tableIdentifier, oldTable.copy(newOptions));
     }
 
+    private Operation convertAlterTableCompact(
+            ObjectIdentifier tableIdentifier,
+            ResolvedCatalogTable resolvedCatalogTable,
+            SqlAlterTableCompact alterTableCompact) {
+        Catalog catalog =
+                catalogManager
+                        .getCatalog(tableIdentifier.getCatalogName())
+                        .orElseThrow(
```

Comment:
orElse(null).
`isManagedTable` accepts nullable catalog.
We can throws exception to a unified mode in below.